src/Client.ts

```diff
@@ -561,3 +550,30 @@ function createApiClientFromOptions(options: ClientOptions): ApiClient {
   const apiUrl = options.apiUrl || ApiUrls[options.env]
   return new ApiClient(apiUrl)
 }
+
+// retrieve a key bundle from given user's contact topic
+async function getUserContactFromNetwork(
+  apiClient: ApiClient,
+  peerAddress: string
+): Promise<PublicKeyBundle | undefined> {
+  // have to avoid undefined to not trip TS's strictNullChecks on recipientKey
+  let recipientKey: PublicKeyBundle | null = null
+  const stream = apiClient.queryIterator(
+    { contentTopics: [buildUserContactTopic(peerAddress)] },
+    { pageSize: 5, direction: SortDirection.SORT_DIRECTION_DESCENDING }
```

Comment:
I'm not sure I fully understand the page size here. Is it necessary to guarantee a PublicKeyBundle will return if they are on the network?

Reply:
So, it's based on the assumption that a winning key bundle will likely be in the first 5 results. It will potentially pull all results down from the network, but will do it 5 at a time, and aborts once it finds a bundle that validates.